paasta_tools/cli/cmds/logs.py

```diff
@@ -549,6 +626,112 @@ def tail_logs(self, service, levels, components, clusters, raw_mode=False):
+        for line in aggregated_logs:
+            print_log(line, levels, raw_mode)
+
+    def filter_and_aggregate_scribe_logs(self, scribe_reader_ctx, scribe_env, stream_name,
+                                         levels, service, components, clusters,
+                                         aggregated_logs, parser_fn=None, filter_fn=None,
+                                         start_time=None, end_time=None):
+        # Scribe specific detail: all messages in app_output get logged at debug level, so we need to add that to
+        # the levels if its one of the components we're grabbing
+        if ('stdout' in components) or ('stderr' in components):
+            levels.append('debug')
+
+        with scribe_reader_ctx as scribe_reader:
+            try:
+                for line in scribe_reader:
+                    if parser_fn:
+                        line = parser_fn(line, clusters, service)
+                    if filter_fn:
+                        if filter_fn(line, levels, service, components, clusters,
+                                     start_time=start_time, end_time=end_time):
+                            aggregated_logs.append(line)
+            except StreamTailerSetupError:
+                log.error("Failed to setup stream tailing for %s in %s" % (stream_name, scribe_env))
+                log.error("Don't Panic! This can happen the first time a service is deployed because the log")
+                log.error("doesn't exist yet. Please wait for the service to be deployed in %s and try again."
+                          % scribe_env)
+                raise
+
+    def scribe_get_from_time(self, scribe_env, stream_name, start_time, end_time):
+        # Scribe connection details
+        host_and_port = scribereader.get_env_scribe_host(scribe_env, True)
+        host = host_and_port['host']
+        port = host_and_port['port']
+
+        # Annoyingly enough, scribe needs special handling if we're trying to retrieve logs from today.
+
+        # There is a datetime.date.today but we can't use it because it isn't UTC
+        today = datetime.datetime.utcnow().date()
+        if start_time.date() == today or end_time.date() == today:
+            # The reason we need a fake context here is because scribereader is a bit incosistent in its
+            # returns. get_stream_reader returns a context that needs to be acquired for cleanup code but
+            # get_stream_tailer simply returns an object that can be iterated over. We'd still like to have
+            # the cleanup code for get_stream_reader to be executed by this function's caller and this is
+            # one of the simpler ways to achieve it without having 2 if statements everywhere that calls
+            # this method
+            @contextmanager
+            def fake_context():
+                yield scribereader.get_stream_tailer(stream_name, host, port, True, -1)
```

Comment:
I don't think this is correct - my understanding is that a stream tailer will only give you logs from now onwards, and if you want logs from earlier today, you'd want get_stream_reader.

e.g. the command `scribereader -e sfo2 -f stream_paasta_yelp-main` would give you logs starting now, but `scribereader -e sfo2 stream_paasta_yelp-main` would give you logs starting today at midnight and ending now.